src/components/Button/index.tsx

```diff
@@ -16,195 +15,154 @@ import themeColors from '@styles/themes/default';
 import CONST from '@src/CONST';
 import validateSubmitShortcut from './validateSubmitShortcut';
 
-const propTypes = {
+type ButtonProps = {
```

Comment:
I think there are two ways of using a Button, either with children or with text and icons:
```ts
import ChildrenProps from '@src/types/utils/ChildrenProps';

type ButtonWithText = {
    /** The text for the button label */
    text: string;

    /** Boolean whether to display the right icon */
    shouldShowRightIcon?: boolean;

    /** The icon asset to display to the left of the text */
    icon?: React.FC<SvgProps> | null;
};

type ButtonProps = (ButtonWithText | ChildrenProps) & { ... }
```

Reply:
This makes destructing a little bit more tricky:
```ts
function Button(
    {
        allowBubble = false,

        iconRight = Expensicons.ArrowRight,
        iconFill = themeColors.textLight,
        iconStyles = [],
        iconRightStyles = [],

        small = false,
        large = false,
        medium = false,

        isLoading = false,
        isDisabled = false,

        onPress = () => {},
        onLongPress = () => {},
        onPressIn = () => {},
        onPressOut = () => {},
        onMouseDown = undefined,

        pressOnEnter = false,
        enterKeyEventListenerPriority = 0,

        style = [],
        innerStyles = [],
        textStyles = [],

        shouldUseDefaultHover = true,
        success = false,
        danger = false,

        shouldRemoveRightBorderRadius = false,
        shouldRemoveLeftBorderRadius = false,
        shouldEnableHapticFeedback = false,

        id = '',
        accessibilityLabel = '',

        ...rest
    }: ButtonProps,
    ref: ForwardedRef<View>,
) {
```

```ts
    const renderContent = () => {
        if ('children' in rest) {
            return rest.children;
        }

        const {text = '', icon = null, shouldShowRightIcon = false} = rest;
```